### PR TITLE
doc: add standard opencontainer docker labels

### PR DESCRIPTION
**Pull request by halkeye**
## Description

Add some of the [standard opencontainer docker labels.](https://github.com/opencontainers/image-spec/blob/main/annotations.md#pre-defined-annotation-keys)

## Motivation and Context

Tools like renovate and dependabot use docker labels to build changelogs when doing an update PR. (https://github.com/halkeye/home-k8s/pull/331 as example). They compare the new docker image to the latest tag, and grab various metadata. If it can, it will also enrich the messages via github releases changelogs.

A working example
```
regctl image config --format '{{ jsonPretty .Config.Labels }}' ghcr.io/plankanban/planka:latest
{
  "org.opencontainers.image.created": "2024-10-02T12:21:17.811Z",
  "org.opencontainers.image.description": "The realtime kanban board for workgroups built with React and Redux.",
  "org.opencontainers.image.licenses": "AGPL-3.0",
  "org.opencontainers.image.revision": "19935e52af46ee95521593af60393861066421ad",
  "org.opencontainers.image.source": "https://github.com/plankanban/planka",
  "org.opencontainers.image.title": "planka",
  "org.opencontainers.image.url": "https://github.com/plankanban/planka",
  "org.opencontainers.image.version": "1.23.1"
}
```

With this PR, I was able to quickly add all but created (as it should be timestamp) and revision (as it should be a sha i think)

```
regctl image config --format '{{ jsonPretty .Config.Labels }}' registry.digitalocean.com/halkeye/oauth2-proxy:v7.7.0-1-g825da4ae-dirty
{
  "org.opencontainers.image.description": "A reverse proxy that provides authentication with Google, Azure, OpenID Connect and many more identity providers.",
  "org.opencontainers.image.licenses": "MIT",
  "org.opencontainers.image.source": "https://github.com/oauth2-proxy/oauth2-proxy",
  "org.opencontainers.image.title": "oauth2-proxy",
  "org.opencontainers.image.url": "https://github.com/oauth2-proxy/oauth2-proxy",
  "org.opencontainers.image.version": "v7.7.0-1-g825da4ae-dirty"
}
```

## How Has This Been Tested?

```
make docker docker-push REGISTRY=registry.digitalocean.com/halkeye DOCKER_BUILD_PLATFORM=linux/amd64 DOCKER_BUILD_PLATFORM_ALPINE=linux/amd64
regctl image config --format '{{ jsonPretty .Config.Labels }}' registry.digitalocean.com/halkeye/oauth2-proxy:v7.7.0-1-g825da4ae-dirty
```

## Checklist:

- [ ] My change requires a change to the documentation or CHANGELOG.
- [ ] I have updated the documentation/CHANGELOG accordingly.
- [x] I have created a feature (non-master) branch for my PR.
- [ ] I have written tests for my code changes.

### CODE REVIEW

**Comment by halkeye** [comment]
I'm not sure how to add tests, or if changelogs are needed

File: Dockerfile [hunk 1]
```diff
@@ -46,7 +46,16 @@ RUN case ${TARGETPLATFORM} in \
 
 # Copy binary to runtime image
 FROM ${RUNTIME_IMAGE}
+ARG VERSION
```
Comment (tuunit) [inline Dockerfile:49]: Duplication. Already defined in line 22.

Please move the `ARG VERSION` from line 22 to the top below:

`ARG RUNTIME_IMAGE=gcr.io/distroless/static:nonroot`

File: Dockerfile [hunk 2]
```diff
@@ -46,7 +46,16 @@ RUN case ${TARGETPLATFORM} in \
...
+
 COPY --from=builder /go/src/github.com/oauth2-proxy/oauth2-proxy/oauth2-proxy /bin/oauth2-proxy
 COPY --from=builder /go/src/github.com/oauth2-proxy/oauth2-proxy/jwt_signing_key.pem /etc/ssl/private/jwt_signing_key.pem
 
+LABEL org.opencontainers.image.licenses=MIT \
+      org.opencontainers.image.description="A reverse proxy that provides authentication with Google, Azure, OpenID Connect and many more identity providers." \
+      org.opencontainers.image.source=https://github.com/oauth2-proxy/oauth2-proxy \
+      org.opencontainers.image.url=https://github.com/oauth2-proxy/oauth2-proxy \
```
Comment (tuunit) [inline Dockerfile:57]:
```suggestion
      org.opencontainers.image.url=https://quay.io/oauth2-proxy/oauth2-proxy \
```

**Review by tuunit — CHANGES_REQUESTED** [review]
As this changes how the build artifacts will look. I would like to see a CHANGELOG entry for it :)

**Comment by tuunit** [comment]
> I'm not sure how to add tests, or if changelogs are needed

The CI build should be enough for testing :)

File: Dockerfile (hunk 2 above)
Comment (halkeye) [reply Dockerfile:57]: ```org.opencontainers.image.url URL to find more information on the image (string)```
shouldn't that be the github readme file and stuff?

File: Dockerfile (hunk 1 above)
Comment (halkeye) [reply Dockerfile:49]: Sure thing, i'll try to get to that after work

**Comment by halkeye** [comment]
> I would like to see a CHANGELOG entry for it :)

can you give me pointers on that? Its just the changelog file right? Maybe something under important notes?

**Comment by tuunit** [comment]
> > I would like to see a CHANGELOG entry for it :)
> 
> can you give me pointers on that? Its just the changelog file right? Maybe something under important notes?

Please just add a single line in the section `Changes since v7.7.0` look at the previous release to check the syntax.

The other sections like important notes etc. are updated by us the maintainers before we publish a release.

File: Dockerfile (hunk 2 above)
Comment (tuunit) [reply Dockerfile:57]: I read it as `image.url`. The url from which to retrieve the image.

`image.source` is the url to the source code hosted on github

`image.documentation` could directly point to the docs

Comment (halkeye) [reply Dockerfile:57]: Renovate (which is the one I care about) only looks at [source](https://docs.renovatebot.com/modules/datasource/docker/) so i'm good with that update

File: Dockerfile [hunk 3]
```diff
@@ -46,7 +47,16 @@ RUN case ${TARGETPLATFORM} in \
...
+
 COPY --from=builder /go/src/github.com/oauth2-proxy/oauth2-proxy/oauth2-proxy /bin/oauth2-proxy
 COPY --from=builder /go/src/github.com/oauth2-proxy/oauth2-proxy/jwt_signing_key.pem /etc/ssl/private/jwt_signing_key.pem
 
+LABEL org.opencontainers.image.licenses=MIT \
+      org.opencontainers.image.description="A reverse proxy that provides authentication with Google, Azure, OpenID Connect and many more identity providers." \
+      org.opencontainers.image.source=https://github.com/oauth2-proxy/oauth2-proxy \
+      org.opencontainers.image.url=https://quay.io/oauth2-proxy/oauth2-proxy \
+      org.opencontainers.image.title=oauth2-proxy \
+      org.opencontainers.image.version=${VERSION}
```
Comment (tuunit) [inline Dockerfile:60]:
```suggestion
LABEL org.opencontainers.image.licenses=MIT \
      org.opencontainers.image.description="A reverse proxy that provides authentication with Google, Azure, OpenID Connect and many more identity providers." \
      org.opencontainers.image.documentation=https://oauth2-proxy.github.io/oauth2-proxy/ \
      org.opencontainers.image.source=https://github.com/oauth2-proxy/oauth2-proxy \
      org.opencontainers.image.url=https://quay.io/oauth2-proxy/oauth2-proxy \
      org.opencontainers.image.title=oauth2-proxy \
      org.opencontainers.image.version=${VERSION}
```

**Review by tuunit — DISMISSED** [review]

**Comment by tuunit** [comment]
Will be merged after the bugfix release for #2802 / #2803

**Review by tuunit — APPROVED** [review]

**Pull request merged**